### PR TITLE
chore: expands zone tests to assert content

### PR DESCRIPTION
**Pull request by kleinfreund**
Creates two new test suites for zone list and detail views that assess their content. Some of the content of the page navigation test was integrated into them and removed from that test.

Removes all snapshot-based tests for zones.

Signed-off-by: Philipp Rudloff <philipp.rudloff@konghq.com>

### CODE REVIEW

File: src/app/zones/views/ZoneListView.spec.ts [hunk 1]
```diff
@@ -15,137 +15,6 @@ describe('ZoneListView', () => {
...
-
-    expect(wrapper.html()).toContain('dpToken')
-
-    await wrapper.find('#insights-tab').trigger('click')
-    expect(wrapper.find('[data-testid="tab-container"]').element).toMatchSnapshot()
-  })
-
   test('updates list when deleting Zone', async () => {
     const firstZoneName = 'zone-1'
     mock('/zones+insights', {
```
[the remark this answers is not in the document]
Comment (kleinfreund) [reply src/app/zones/views/ZoneListView.spec.ts:20]: Not sure, but it's perfectly fine to keep this as a unit test

File: features/zones/Create.feature [hunk 2]
```diff
@@ -68,5 +68,6 @@ Feature: The create Zone flow works
         zoneInsight:
           subscriptions:
             - connectTime: '2020-07-28T16:18:09.743141Z'
+              status: {}
```
Comment (johncowen) [inline features/zones/Create.feature:71]: How comes we added this here? Is `status` always an empty object immediately on zone creation? I also noticed we do this through this PR, does doing that have any consequences for what we are covering in the test?

**Review by johncowen — APPROVED** [review]
LGTM!

I had a question just out of curiosity, but all looks fine to me.

File: src/app/zones/views/ZoneListView.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (johncowen) [reply src/app/zones/views/ZoneListView.spec.ts:20]: Well I think my point is its not really a unit test is it? Its not really testing a single isolated unit of source code, rather the a single feature of the entire application's code base.

I looked at it and didn't see why it wouldn't be possible, so just wondered, we can move this to where it should be in a follow up though.

File: features/zones/Create.feature (hunk 2 above)
Comment (kleinfreund) [reply features/zones/Create.feature:71]: There some pages that access this object (via `SubscriptionDetails`) and setting this as an empty object is the most minimal way of getting the errors out of those tests.

Comment (johncowen) [reply features/zones/Create.feature:71]: So when you say "getting the errors out of those tests" do you mean "getting errors out" as in making the thing error, or making sure it doesn't error? I think its the latter but just checking

Comment (kleinfreund) [reply features/zones/Create.feature:71]: Making sure the application code doesn't throw an exception on account of accessing `status.property` (as `status` would be wholly absent from the mock data without this change).

Comment (johncowen) [reply features/zones/Create.feature:71]: Hmm I might catch up with you elsewhere on this one, just so I follow completely.

**Pull request merged**